scripts/src/main/scala/org/dbpedia/extraction/scripts/NormalizeDatasets.scala

```diff
@@ -0,0 +1,280 @@
+  }
+
+  // URI prefix for wikidata entities
+  private val WikidataResource = "http://wikidata.dbpedia.org/resource/Q"
+
+  def main(args: Array[String]): Unit = {
+    require(args != null && args.length >= 6,
+      "need at least six args: " +
+        /*0*/ "extraction config file" +
+        /*1*/ "prefix of mapping file (eg. wikidata)"+
+        /*2*/ "comma-separated names of datasets mapping URIs to wikidata identifiers (eg. wikidata-sameas)" +
+        /*3*/ "mapping file suffix (e.g. '.nt.gz', '.ttl', '.ttl.bz2'), " +
+        /*4*/ "comma-separated names of input datasets (e.g. 'infobox-properties,mappingbased-properties'), " +
+        /*5*/ "output dataset name extension (e.g. '-normalized'), ")
+
+
+    val config = ConfigUtils.loadConfig(args(0), "UTF-8")
+
+    val baseDir = ConfigUtils.getValue(config, "base-dir", true)(new File(_))
+    if (!baseDir.exists)
+      throw new IllegalArgumentException("dir " + baseDir + " does not exist")
+    val langConfString = ConfigUtils.getString(config, "languages", false)
+    val languages = ConfigUtils.parseLanguages(baseDir, Seq(langConfString))
+    require(languages.nonEmpty, "no languages")
+
+    val formats = parseFormats(config, "uri-policy", "format")
+
+    val mappingPrefix = args(1)
+
+    val mappings = split(args(2))
+    require(mappings.nonEmpty, "no mapping datasets")
+
+    // Suffix of mapping files, for example ".nt", ".ttl.gz", ".nt.bz2" and so on.
+    // This script works with .nt, .ttl, .nq or .tql files, using IRIs or URIs.
+    val mappingSuffix = args(3)
+    require(mappingSuffix.nonEmpty, "no mapping file suffix")
+
+    val inputs = split(args(4))
+    require(inputs.nonEmpty, "no input datasets")
+
+    val outputExtension = args(5)
+    require(outputExtension.nonEmpty, "no output name extension")
+
+    val extensions = List(outputExtension, outputExtension + "-subject-rejected", outputExtension + "-object-rejected", outputExtension + "-rejected")
+
+    // Suffixes of input/output files, for example ".nt", ".ttl.gz", ".nt.bz2" and so on.
+    // This script works with .nt, .ttl, .nq or .tql files, using IRIs or URIs.
+    val suffixes = formats.keys.map("." + _)
+    require(suffixes.nonEmpty, "no input/output file suffixes")
+
+    // We really want to saturate CPUs and disk, so we use 50% more workers than CPUs
+    val jobs = for(language <- languages) yield {
+      val wikiFinder = new Finder(baseDir, language, "wiki") // Finds Wikipedia extracted datasets
+      val date = wikiFinder.dates().last
+      val datasets = for(input <- inputs; extension <- extensions) yield new Dataset(input + extension)
+      val destination = createDestination(wikiFinder, date, formats, datasets)
+      val mappingFinder = new DateFinder(baseDir, Language(mappingPrefix)) // Finds wikidata mapping dataset
+
+      val map = {
+        // Try reading from cache
+        val cache = wikiFinder.file(date, "sameas-mappings.obj")
```

Comment:
if the script is re-started with another language withouth deleting the file it will produce false results